gesos-image-build.pipeline

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env groovy
+
+@Library(['PPCmanifest','security-ci-commons-shared-lib']) _
+
+def repoStatusReportUrl
+def gitCommitStatusContext = 'GE SOS Image Build'
+
+pipeline {
+    agent {
+        docker {
+            image 'openjdk:8'
+            label 'dind'
+        }
+    }
+    parameters {
+        string(
+            name: 'IAM_CONTAINER_CONFIG_BRANCH_NAME',
+            defaultValue: BRANCH_NAME,
+            description: 'iam-container-config repo branch name to use for GE SOS image build'
+        )
+        string(
+            name: 'IAM_CONTAINER_CONFIG_FOLDER_NAME',
+            defaultValue: 'uaa',
+            description: 'Name of container config folder with Dockerfile and prebuild script in the ' +
+                         'iam-container-config repo to use for GE SOS image build'
+        )
+        string(
+            name: 'IMAGE_TAG',
+            defaultValue: 'BUILD_TAG_FROM_BRANCH_NAME',
+            description: 'Tag for docker image to push to ECR'
+        )
+        string(
+            name: 'IMAGE_NAME',
+            defaultValue: 'iam-uaa',
+            description: 'Name for docker image to push to ECR'
+        )
+    }
+    environment {
+        GESOS_API_KEY = credentials('GESOS_API_KEY')
+        GITHUB_CREDENTIALS = credentials('GITHUB_CREDENTIALS')
+    }
+    stages {
+        stage('Github status configuration') {
+            steps {
+                script {
+                    commitSha = sh(script: "git rev-parse HEAD", returnStdout: true).trim()
+                    repoStatusReportUrl = sh(script: 'git config --get remote.origin.url', returnStdout: true).trim()
+                                                                                                              .replaceFirst('^https://([^/]+)/([^.]+)(\\.git)?$',
+                                                                                                                            'https://$1/api/v3/repos/$2/statuses/' + commitSha)
+                }
+            }
+        }
+        stage('Trigger GE SOS build pipeline') {
+            steps {
+                script {
+                    setBuildStatus(repoStatusReportUrl, gitCommitStatusContext, 'in progress', 'pending')
+                    iamContainerConfigBranchName = params.IAM_CONTAINER_CONFIG_BRANCH_NAME
+                    if (iamContainerConfigBranchName == BRANCH_NAME) {
+                        iamContainerConfigBranchName = BRANCH_NAME.replaceAll('rc_', '').replaceAll('release_', '')
```

Comment:
The job should be configured for rc, release and PRs branches only.  There is no reason to build GESOS from predix_extensions.

rc and release branches publish UAA war to two different locations: MAAXA-SNAPSHOT and MAAXA respectively.  How are we gonna handle this? Should we have different branches in iam-container-config, one for RC and one for release? Or should we use one branch in iam-container-config and once rc is promoted to release change pre-build script?  

Maybe we should have two branches in iam-container-config one for UAA rc and one for UAA release?Then above if is not needed.

Reply:
Done, configured job for rc, release branches only and will create separate branch in  iam-container-config repo. For now, only created rc_75.18.1 branch and manually triggering gesos build pipeline using this branch.